dejavu_bot.py

```diff
@@ -1,111 +1,152 @@
 """
 This is all of the code for a discord bot that will find a random message in
-your channel history and send it formatted in an image with a background
+your channel history and send it in either text or image format and there is also
+a game where you have to guess who said the message
 
-invoke with `/dejavu`
+Invoke with `/dejavu`
+Arguments: text, image, whosaid
 """
 
 import os
 from datetime import datetime, timedelta, timezone
 from io import BytesIO
-from random import choice, randrange
+from random import choices, randrange
 
 from PIL import Image, ImageColor, ImageDraw, ImageFont
 
 import discord
-from discord.ext import commands
+from discord import app_commands
 
+# Load .env file
 from dotenv import load_dotenv
+
 load_dotenv()
 
+# Set up bot
 intents = discord.Intents.default()
 intents.message_content = True
+bot = discord.Client(intents=intents)
 
-bot = commands.Bot(command_prefix='/', intents=intents)
+# Set up slash commands
+tree = app_commands.CommandTree(bot)
 
+# Colors for the image mode.
 VERY_DARK_COLORS = [
-    'black',
-    'darkblue',
-    'darkmagenta',
-    'darkslategrey',
-    'indigo',
-    'midnightblue',
-    'navy',
-    'purple'
+    "black",
+    "darkblue",
+    "darkmagenta",
+    "darkslategrey",
+    "indigo",
+    "midnightblue",
+    "navy",
+    "purple",
 ]
 
-bot.who_said_context = None
-
-@bot.command()
-async def dejavu(ctx, arg):
+# Initalize whosaid game variables
+bot.whosaid = {}
+bot.whosaid["playing"] = False
+bot.whosaid["channel"] = None
+bot.whosaid["author"] = None
+bot.whosaid["message"] = None
+bot.whosaid["second_chance"] = False
+
+# Command setup and argument definition.
+@tree.command(
+    name="dejavu",
+    description="Devjavu bot",
+)
+@app_commands.choices(
+    choices=[
+        app_commands.Choice(name="Retrieve a random message.", value="text"),
+        app_commands.Choice(
+            name="Retrieve a random message and put it in an image.", value="image"
+        ),
+        app_commands.Choice(
+            name="Retrieve a random message and you must guess who said it by mentioning them.",
+            value="whosaid",
+        ),
+    ]
+)
+async def dejavu(inter, choices: app_commands.Choice[str]):
```

Comment:
`choices` needs to have a different name because we import `random.choices`

Reply:
:+1:

Reply:
:+1: